hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/TrashOzoneFileSystem.java

```diff
@@ -0,0 +1,418 @@
+        LOG.trace("Iterating directory: {}", pathKey);
+        OFSPath ofsPath = new OFSPath(pathKey);
+        String ofsPathPrefix =
+            ofsPath.getNonKeyPathNoPrefixDelim() + OZONE_URI_DELIMITER;
+        while (keyIterator.hasNext()) {
+          Table.KeyValue< String, OmKeyInfo > kv = keyIterator.next();
+          String keyPath = ofsPathPrefix + kv.getValue().getKeyName();
+          LOG.trace("iterating key path: {}", keyPath);
+          String k = kv.getKey();
+          if (!kv.getValue().getKeyName().equals("")
+              && kv.getKey().startsWith("/" + pathKey)) {
+            keyPathList.add(keyPath);
+          }
+          if (keyPathList.size() >= OZONE_FS_ITERATE_BATCH_SIZE) {
+            if (!processKeyPath(keyPathList)) {
+              return false;
+            } else {
+              keyPathList.clear();
+            }
+          }
+        }
+        if (keyPathList.size() > 0) {
+          if (!processKeyPath(keyPathList)) {
+            return false;
+          }
+        }
+        return true;
+      } else {
+        LOG.trace("iterating file: {}", path);
+        keyPathList.add(pathKey);
+        return processKeyPath(keyPathList);
+      }
+    }
+
+    FileStatus getStatus() {
+      return status;
+    }
+  }
+
+  private class RenameIterator extends OzoneListingIterator {
+    private final String srcPath;
+    private final String dstPath;
+
+    RenameIterator(Path srcPath, Path dstPath)
+        throws IOException {
+      super(srcPath);
+      this.srcPath = pathToKey(srcPath);
+      this.dstPath = pathToKey(dstPath);
+      LOG.trace("rename from:{} to:{}", this.srcPath, this.dstPath);
+    }
+
+    @Override
+    boolean processKeyPath(List<String> keyPathList) throws IOException {
+      for (String keyPath : keyPathList) {
+        String newPath = dstPath.concat(keyPath.substring(srcPath.length()));
+        LOG.info(newPath);
+        OFSPath src = new OFSPath(keyPath);
+        OFSPath dst = new OFSPath(newPath);
+
+        String volumeName = src.getVolumeName();
+        String bucketName = src.getBucketName();
+        String keyName = src.getKeyName();
+
+        OzoneManagerProtocolProtos.KeyArgs keyArgs =
+            OzoneManagerProtocolProtos.KeyArgs.newBuilder().setKeyName(keyName)
+            .setVolumeName(volumeName).setBucketName(bucketName).build();
+
+        String toKeyName = dst.getKeyName();
+
+        OzoneManagerProtocolProtos.RenameKeyRequest renameKeyRequest =
+            OzoneManagerProtocolProtos.RenameKeyRequest.newBuilder()
+            .setKeyArgs(keyArgs).setToKeyName(toKeyName).build();
+        OzoneManagerProtocolProtos.OMRequest omRequest =
+            OzoneManagerProtocolProtos.OMRequest.newBuilder()
+            .setClientId(UUID.randomUUID().toString())
+            .setRenameKeyRequest(renameKeyRequest)
+            .setCmdType(OzoneManagerProtocolProtos.Type.RenameKey).build();
+        try {
+          ozoneManager.getOmServerProtocol().
```

Comment:
Lets not use the rpc sever here and lets use the following call in
` 
@Override
  public OMResponse submitRequest(RpcController controller,
      OMRequest request) throws ServiceException {

    return dispatcher.processRequest(request, this::processRequest,
        request.getCmdType(), request.getTraceID());
  }
`

Reply:
https://github.com/apache/ozone/blob/b0d54d41b51504359373be541b6a337ce9aa2c52/hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/protocolPB/OzoneManagerProtocolServerSideTranslatorPB.java#L119

Reply:
i am calling the same method here i.e 
` public OMResponse submitRequest(RpcController controller,
      OMRequest request) throws ServiceException {`. om.getOmServerProtocol() returns `OzoneManagerProtocolServerSideTranslatorPB` object.